tests/provision_test.go

```diff
@@ -51,62 +51,103 @@ func exhaustiveVolumeTests(parameters map[string]string) {
-	verifySnapshotCreated(snapName)
-	createClone(clonePvcName, snapName, scObj.Name)
-	By("Creating and deploying clone app pod", createDeployVerifyCloneApp)
+	createSnapshot(pvcNameFS, snapNameFS)
+	verifySnapshotCreated(snapNameFS)
+	createClone(clonePvcNameFS, snapNameFS, scObj.Name)
+	By("Creating and deploying clone app pod", func() { createDeployVerifyCloneApp(cloneAppNameFS, clonePvcNameFS) })
 }
 
 // Removes the snapshot/clone resources
 func snapshotAndCloneCleanUp() {
-	deleteAppDeployment(cloneAppName)
-	deletePVC(clonePvcName)
-	deleteSnapshot(pvcName, snapName)
+	deleteAppDeployment(cloneAppNameFS)
+	deletePVC(clonePvcNameFS)
+	deleteSnapshot(pvcNameFS, snapNameFS)
 }
 
 // Removes the resources
 func cleanUp() {
-	deleteAppDeployment(appName)
-	deletePVC(pvcName)
+	deleteAppDeployment(appNameFS)
+	deletePVC(pvcNameFS)
 	By("Deleting storage class", deleteStorageClass)
 }
 
 func blockVolCreationTest() {
 	By("Creating default storage class", createStorageClass)
-	By("creating and verifying PVC bound status", createAndVerifyBlockPVC)
+	By("creating and verifying PVC bound status", func() { createAndVerifyPVC(pvcNameBlock) })
 
-	By("Creating and deploying app pod", createDeployVerifyBlockApp)
+	By("Creating and deploying app pod", func() { createDeployVerifyApp(appNameBlock, pvcNameBlock) })
 	By("verifying ZFSVolume object", VerifyZFSVolume)
 	By("verifying ZFSVolume property change", VerifyZFSVolumePropEdit)
-	By("Deleting application deployment")
 
-	createSnapshot(pvcName, snapName)
-	verifySnapshotCreated(snapName)
-	createClone(clonePvcName, snapName, scObj.Name)
-	By("Creating and deploying clone app pod", createDeployVerifyCloneApp)
+	createSnapshot(pvcNameBlock, snapNameBlock)
+	verifySnapshotCreated(snapNameBlock)
+	createClone(clonePvcNameBlock, snapNameBlock, scObj.Name)
+	By("Creating and deploying clone app pod", func() { createDeployVerifyCloneApp(cloneAppNameBlock, clonePvcNameBlock) })
 
-	By("Deleting clone and main application deployment")
-	deleteAppDeployment(cloneAppName)
-	deleteAppDeployment(appName)
+	By("Deleting main application deployment")
+	deleteAppDeployment(appNameBlock)
 
-	By("Deleting snapshot, main pvc and clone pvc")
-	deletePVC(clonePvcName)
-	deleteSnapshot(pvcName, snapName)
-	deletePVC(pvcName)
+	zvName := getZVName(pvcNameBlock)
+	By("Deleting main pvc")
+	deletePVC(pvcNameBlock)
 
+	By("Verifying ZFSVolume object after pvc deletion when snapshot is present", VerifyZFSVolume)
+
+	By("Deleting clone application deployment")
+	deleteAppDeployment(cloneAppNameBlock)
+
+	By("Deleting snapshot and clone pvc")
+
+	deletePVC(clonePvcNameBlock)
+	By("Verifying that ZV is present after pvc deletion ", func() { IsZVPresentConsistently(zvName) })
+	deleteSnapshot(pvcNameBlock, snapNameBlock)
+	By("Verifying that ZV is deleted after snapshot deletion ", func() { IsZVDeletedEventually(zvName) })
+
+	By("Deleting storage class", deleteStorageClass)
+}
+
+func blockVolCreationWithReclaimRetainTest() {
```

Comment:
If this is just for checking the retain, then we don't need any applications deployed?

Reply:
added the code to create a pv and pv and check if the pvc is bound . Havent deployed the app

Reply:
I mis understood your comment . I thought you need the application to be deployed from manully created PVC which comes from retained ZV, so i added creating a manual pv and pvc from retained ZV. 

Any ways i have remove the app deployment code in this test as this was redundant, but the snapshot test exist as we want to verify that ZV is present even after snapshot is deleted.